worksheet_A/TerminalHacking.cpp

```diff
@@ -41,6 +53,75 @@ int main()
 
 	// TODO: implement the rest of the game
 
+	// Initialise variables
+	std::string guess;
+	int maxLives = 4;
+	int remainingLives = maxLives;
+
+	// Two different ways of displaying text to console
+	std::cout << "Guess the secret word." << std::endl;
+	printf("You have %i guesses. \n\n", maxLives);
+
+	while (remainingLives > 0)
+	{
+		bool isValidGuess = false;
+
+		// Recieves player guess and checks it is in the list
+		while (!isValidGuess)
+		{
+			printf("Enter your guess: ");
+			std::cin >> guess;
+			std::cout << "your guess was: " << guess << std::endl;
+
+			// Scan through the line, converting each character to upper case.
+			for (int i = 0; i < guess.length(); i++)
+			{
+				if (isalpha(guess[i]))
+				{
+					guess[i] = toupper(guess[i]);
+				}
+			}
+
+			// Check guess is in list
```

Comment:
Can also be done in one line using built-in C++ functions -- see http://stackoverflow.com/questions/1701067/how-to-check-that-an-element-is-in-a-stdset

In this case the one-line solution also scales better -- O(log n) rather than O(n)